src/pages/resultsView/ResultsViewPage.tsx

```diff
@@ -423,7 +423,7 @@ export default class ResultsViewPage extends React.Component<IResultsViewPagePro
 
                 {
                     (this.resultsViewPageStore.studies.isComplete) && (
-                        <MSKTabs key={(window as any).routingStore.queryHash} activeTabId={this.currentTab(this.props.params.tab)} unmountOnHide={false}
+                        <MSKTabs key={this.resultsViewPageStore.queryHash} activeTabId={this.currentTab(this.props.params.tab)} unmountOnHide={false}
```

Comment:
the problem is that if page children observe query directly, then they react to a route switch (e.g. clicking cbio logo to return to homepage) before Page has unmounted.  So for example, if on results page, you click cbio logo, the oncoprint would re-render because the query has "changed"
this all deserves more thought but this works.

Reply:
i say deserves more thought because its a pretty big restriction to say nothing inside page can directly observe routingStore.